Assets/_Functionalities/Wms/Scripts/ATMLayerManager.cs

```diff
@@ -0,0 +1,112 @@
+{
+    public class ATMLayerManager : IDisposable
+    {
+        private ATMTileDataLayer[] ATMTileDataLayers => atmTileDataLayers;
+
+        private int zoomLevel = -1;
+        private int lastZoomLevel = -1;
+        private ATMDataController timeController;
+        private ATMTileDataLayer[] atmTileDataLayers;
+
+        public ATMLayerManager(ATMDataController timeController)
+        {
+            this.timeController = timeController;
+            timeController.ChangeYear += OnYearChanged;
+        }
+
+        private void OnYearChanged(int a)
+        {
+            UpdateCurrentZoomLayer(true);
+        }
+
+        public void SwitchLayerToCurrentZoomLevel(bool force = false)
+        {
+            zoomLevel = timeController.CalculateZoomLevel();
+
+            UpdateCurrentZoomLayer(force);
+        }
+
+        public void UpdateCurrentZoomLayer(bool force = false)
+        {
+            if (ATMTileDataLayers == null) return;
+            if (zoomLevel == lastZoomLevel && !force) return;
+
+            //get the current enabled zoom layer and update it with setvisibletilesdirty
+            int index = timeController.GetZoomLayerIndex(zoomLevel);
+
+            ActivateLayer(zoomLevel);
+            if (index >= 0 && index < ATMTileDataLayers.Length)
+            {
+                ATMTileDataLayers[index].SetVisibleTilesDirty();
+            }
+
+            lastZoomLevel = zoomLevel;
+        }
+
+        public void CreateTileHandlerForEachZoomLevel(Transform parent, ATMMapLayerGameObject layerGameObjectPrefab)
+        {
+            var zoomBounds = timeController.GetZoomBoundsAllYears();
+            int numberOfZoomLevels = zoomBounds.y - zoomBounds.x + 1;
+            atmTileDataLayers = new ATMTileDataLayer[numberOfZoomLevels];
+            for (int i = 0; i < ATMTileDataLayers.Length; i++)
+            {
+                CreateTileLayerForDataLayer(parent, i, zoomBounds.x + i, layerGameObjectPrefab);
+            }
+        }
+
+        private void ActivateLayer(int zoomLevel)
+        {
+            int index = timeController.GetZoomLayerIndex(zoomLevel);
+            for (int i = 0; i < ATMTileDataLayers.Length; i++)
+            {
+                ATMTileDataLayers[i].isEnabled = i == index;
+                
+                if (i != index)
+                {
+                    ATMTileDataLayers[i].CancelTiles();
+                }
+            }
+
+            // I know .. reflection. But this will ensure that if another zoom level enables, that the tilesizes doesn't
+            // go mad. And trying to disable and enable all gameobjects that have a cartesian tile layer seems like
+            // overkill and might have issues due to onEnable and onDisable's triggering 
+            MethodInfo privateMethod = typeof(CartesianTiles.TileHandler).GetMethod("GetTilesizes", BindingFlags.NonPublic | BindingFlags.Instance);
+            if (privateMethod != null)
+            {
```

Comment:
Heb je hier geprobeerd de TileHandler op te vragen en dan:
tilehandler.RemoveLayer(ATMTileDataLayers[0];
tilehandler.AddLayer(ATMTileDataLayers[0];

Dan ververst de tilehandler als het goed is met de juiste tilesizes en hoef je geen reflection te gebruiken. Want met uitvoerig testen en dit helemaal na te lopen bleek dat de oneable en ondisable geen problemen waren.

Reply:
aangezien je in de volgende method de tilesize bij creatie zet, zou dit moeten werken, zo heeft het eerder ook gewerkt zonder problemen

Reply:
Ik had redenen, maar ik ga het niet nu veranderen. Teveel risico dat wel optreed wat er gebeurde en dit is weliswaar niet optimaal, maar werkt wel en geeft ook geen problemen (zeker niet in een POC)